wasm/src/main/scala/ir2wasm/WasmExpressionBuilder.scala

```diff
@@ -2019,4 +2020,50 @@ private class WasmExpressionBuilder private (
     }
     t.tpe
   }
+
+  private def genMatch(tree: IRTrees.Match): IRTypes.Type = {
+    val IRTrees.Match(selector, cases, defaultBody) = tree
+    val selectorLocal = fctx.addSyntheticLocal(TypeTransformer.transformType(selector.tpe)(ctx))
+    genTreeAuto(selector)
+    instrs += LOCAL_SET(selectorLocal)
+
+    fctx.block(TypeTransformer.transformType(tree.tpe)(ctx)) { doneLabel =>
+      fctx.block() { defaultLabel =>
+        val caseLabels = cases.map(c => c._1 -> fctx.genLabel())
+        for (caseLabel <- caseLabels)
+          instrs += BLOCK(BlockType.ValueType(), Some(caseLabel._2))
+
+        for {
+          caseLabel <- caseLabels
+          matchableLiteral <- caseLabel._1
+        } yield {
```

Comment:
This should be a `foreach`:
```suggestion
        } {
```